it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.datastream;
+
+import com.google.cloud.datastream.v1.*;
+import com.google.protobuf.Duration;
+import com.google.protobuf.FieldMask;
+import org.apache.arrow.util.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Default class for implementation of {@link DatastreamResourceManager} interface.
+ */
+public final class DefaultDatastreamResourceManager implements DatastreamResourceManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultDatastreamResourceManager.class);
+    private static final String AVRO_FILE_FORMAT = "avro";
+    private static final String FIELD_STATE = "state";
+    private final DatastreamClient datastreamClient;
+    private final String location;
+    private final String projectId;
+    private final Set<String> createdStreamIds;
+    private final Set<String> createdConnectionProfileIds;
+
+
+    private DefaultDatastreamResourceManager(Builder builder) throws IOException {
+        this(
+                DatastreamClient.create(),
+                builder.location,
+                builder.projectId);
+    }
+
+    @VisibleForTesting
+    public DefaultDatastreamResourceManager(
+            DatastreamClient datastreamClient,
+            String location,
+            String projectId) {
+        this.datastreamClient = datastreamClient;
+        this.location = location;
+        this.projectId = projectId;
+        this.createdStreamIds = Collections.synchronizedSet(new HashSet<>());
+        this.createdConnectionProfileIds = Collections.synchronizedSet(new HashSet<>());
+    }
```

Comment:
I made all the GCP RM's use a similar structure to this before adding the non-GCP RM's, but I prefer the way we have been making the non-GCP RM's that pass the builder to the testing constructor rather than individual variables. I find it cleaner and doesn't restrict which builder variables are set in unit tests. I will go back and refactor the GCP RM's too most likely.
```suggestion
    private DatastreamResourceManager(Builder builder) throws IOException {
        this(
                DatastreamClient.create(),
                builder);
    }

    @VisibleForTesting
    DatastreamResourceManager(
            DatastreamClient datastreamClient,
            DatastreamResourceManager.Builder builder) {
        this.datastreamClient = datastreamClient;
        this.location = builder.location;
        this.projectId = builder.projectId;
        this.createdStreamIds = Collections.synchronizedSet(new HashSet<>());
        this.createdConnectionProfileIds = Collections.synchronizedSet(new HashSet<>());
    }
```